pkg/cacheutil/memcached_client.go

```diff
@@ -272,9 +272,13 @@ func (c *memcachedClient) SetAsync(ctx context.Context, key string, value []byte
 }
 
 func (c *memcachedClient) GetMulti(ctx context.Context, keys []string) map[string][]byte {
+	if len(keys) == 0 {
+		return nil
+	}
+
 	batches, err := c.getMultiBatched(ctx, keys)
 	if err != nil {
-		level.Warn(c.logger).Log("msg", "failed to fetch items from memcached", "err", err)
+		level.Warn(c.logger).Log("msg", "failed to fetch items from memcached", "num_keys", len(keys), "first_key", keys[0], "err", err)
```

Comment:
```suggestion
		level.Warn(c.logger).Log("msg", "failed to fetch items from memcached", "numKeys", len(keys), "firstKey", keys[0], "err", err)
```
Maybe it's silly but we stick to camel Case everywhere in Thanos in logs for keys.. (:

Reply:
Sure! I wasn't aware of that. Now I do :)